File: bridge/StereoPannerNode.cpp
#include "./dart_api/dart_api.h"
#include "LabSound/LabSound.h"
#include "KeepNode.cpp"
using namespace lab;

DART_EXPORT int createStereoPannerNode(AudioContext* context) {
    auto node = std::make_shared<StereoPannerNode>(*context);
    return keepNode(node);
}

DART_EXPORT int StereoPannerNode_pan(int nodeId) {
    auto node = std::static_pointer_cast<StereoPannerNode>(getNode(nodeId));
    return node ? keepAudioParam(nodeId, 0, node->pan()) : -1;
}

File: lab_sound_ffi/lib/src/lab_sound/stereo_panner_node.dart
import 'audio_context.dart';
import 'audio_node.dart';
import 'audio_param.dart';
import 'lab_sound.dart';

class StereoPannerNode extends AudioNode {
  late AudioParam delayTime;
  StereoPannerNode(AudioContext ctx): super(ctx, LabSound().createStereoPannerNode(ctx.pointer)) {
    delayTime = AudioParam(this.ctx, nodeId, LabSound().StereoPannerNode_pan(nodeId));
  }
}